File: src/main/java/io/github/chaosdave34/kitpvp/abilities/impl/magician/LevitateAbility.java
package io.github.chaosdave34.kitpvp.abilities.impl.magician;

import io.github.chaosdave34.kitpvp.ExtendedPlayer;
import io.github.chaosdave34.kitpvp.abilities.Ability;

import net.kyori.adventure.text.Component;
import org.bukkit.entity.Player;
import org.jetbrains.annotations.NotNull;

import java.util.List;

public class LevitateAbility extends Ability {
    public LevitateAbility() {
        super("levitate", "Levitate", Type.RIGHT_CLICK, 25);
    }


    @Override
    public @NotNull List<Component> getDescription() {
        return createSimpleDescription(
                "Let the players around you",
                "float into air."
        );
    }

    @Override
    public boolean onAbility(Player p) {
        p.getNearbyEntities(10, 10, 10).forEach(entity -> {
            if (entity instanceof Player target) {
                if (ExtendedPlayer.from(target).inGame()) {
                    target.setVelocity(target.getVelocity().setY(5));
                }
            }
        });
        return true;
    }
}

File: src/main/kotlin/io/github/chaosdave34/kitpvp/abilities/impl/magician/LevitateAbility.kt
package io.github.chaosdave34.kitpvp.abilities.impl.magician

import io.github.chaosdave34.kitpvp.abilities.Ability
import net.kyori.adventure.text.Component
import org.bukkit.entity.Player

class LevitateAbility : Ability("levitate", "Levitate", Type.RIGHT_CLICK, 20) {
    override fun getDescription(): List<Component> = createSimpleDescription("Let all enemies around you float into air.")

    override fun onAbility(player: Player): Boolean {
        player.getNearbyEntities(10.0, 10.0, 10.0).forEach { target ->
            if (target.checkTargetIfPlayer()) {
                target.velocity = target.velocity.setY(5)
            }
        }
        return true
    }
}